### PR TITLE
Create Email-Communication-Readme

### PR DESCRIPTION
**Pull request by altmannyarden**
## Status
In progress 

## Description
Added a readme description of the pack configurations

### CODE REVIEW

File: content-repo/extra-docs/articles/Email-Communication-Readme [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+id: EmailCommunication
```
Comment (glicht) [inline content-repo/extra-docs/articles/Email-Communication-Readme:2]: Let's use the following naming conventino:
```suggestion
id: email-communication
```

**Comment by ShirleyDenkberg** [comment]
@glicht @michalgold Doc review completed.

File: content-repo/extra-docs/articles/Email-Communication-Readme.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+|------------------ | ------------- |
+| Add CC to email | Add 1 or more CC recipients to the email as a comma-separated list of email addresses.  |
+| Message body | Write the body of the email reply. It is sent as an email and added to the War Room entry. |
+| Attachments | Add attachments to the email reply.  |
+| Email thread | Displays the entire email thread including the original email and all email replies. |  
+| Original Email HTML | Displays in HTML format the original email that opened the incident. |  
+| Mail Attachments | Displays the metadata of the email attachments. |  
+ ---
+
+> Warning: There is no notification for failed emails/addresses.
```
Comment (michalgold) [inline content-repo/extra-docs/articles/Email-Communication-Readme.md:77]: what does this mean? @altmannyarden @ShirleyDenkberg

Comment (altmannyarden) [reply content-repo/extra-docs/articles/Email-Communication-Readme.md:77]: @ShirleyDenkberg, please remove this line. In other mail clients, you don't get on error about the wrong email address (you will get another email from the mail server that the email address wasn't found)

File: content-repo/extra-docs/articles/Email-Communication-Readme.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+description: Communication across and between departments is a vital component of collecting information, and managing and remediating security events. The Email Communication content pack enables security teams to automate and streamline the communication and notification process with users across your organization via email.
+ 
+---
+This Email Communication pack enables security teams to reply to an email as part of an incident. The SOC team is able to communicate directly through the Cortex XSOAR platform as part of the remediation process. 
+
+
+
+
+## Pack Workflow
+When an email is sent to the email address configured in your email integration, your email listener fetches the incoming email in Cortex XSOAR.
```
Comment (altmannyarden) [inline content-repo/extra-docs/articles/Email-Communication-Readme.md:13]: Need to change the word hash to number because we're generating a random number and not creating a hash.

File: content-repo/extra-docs/articles/Email-Communication-Readme.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+
+
+
+## Pack Workflow
+When an email is sent to the email address configured in your email integration, your email listener fetches the incoming email in Cortex XSOAR. 
+
+If the email is in response to an existing incident, an 8-digit hash will appear next to the subject of the email, for example <93075875> Incident1.
+
+The pre-process rule searches for the 8-digit hash in the subject of the email to link to the incident.
```
Comment (altmannyarden) [inline content-repo/extra-docs/articles/Email-Communication-Readme.md:17]: The pre-process script. The rule only runs the script

File: content-repo/extra-docs/articles/Email-Communication-Readme.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+
+## In This Pack
+The Email Communication content pack includes several content items.
+
+### Automations
+There are 3 Automations in this pack.
+* [DisplayEmailHtml](https://xsoar.pan.dev/docs/reference/scripts/display-email-html): Displays the original email in HTML format in the incident layout.
+
+* [PreprocessEmail](https://xsoar.pan.dev/docs/reference/scripts/preprocess-email): Pre-process script for Email Communication layout. This script checks if the incoming email contains an Incident ID to link the mail to an existing incident, and tags the email as "email-thread".
```
Comment (altmannyarden) [inline content-repo/extra-docs/articles/Email-Communication-Readme.md:36]: the pre-process now checks for 8-digit number and not the incident ID

File: content-repo/extra-docs/articles/Email-Communication-Readme.md (hunk 2 above)
Comment (kirbles19) [reply content-repo/extra-docs/articles/Email-Communication-Readme.md:77]: We will reword it a bit, but this should stay.
The whole point of this pack is to replace having to use a mail client as part of an incident. If we are not notifying users of an email that wasn't sent successfully, we're requiring them to rely on the mail client still.

These are the exact kinds of information that lead to CS tickets.

**Review by kirbles19 — APPROVED** [review]

File: content-repo/extra-docs/articles/Email-Communication-Readme.md (hunk 2 above)
Comment (ShirleyDenkberg) [reply content-repo/extra-docs/articles/Email-Communication-Readme.md:77]: I will reword this as follows:
Note: If an email cannot be sent to a specified address, no notification will appear in Cortex XSOAR.

**Comment by kirbles19** [comment]
@altmannyarden Done. Implemented all comments.

**Comment by omercnet** [comment]
there's no mentioning of having to customize `service_mail` parameter in the `Send Reply` button which is required as part of setting up the pack, which you cannot really do without copying the layout

see https://github.com/demisto/etc/issues/31911

**Comment by omercnet** [comment]
this content pack also requires configuring Demisto REST API, should be documented

**Comment by ShirleyDenkberg** [comment]
@omercnet Thanks for your comments.  I implemented them.  Please review.
@glicht @michalgold I implemented Omer's comments.

File: content-repo/extra-docs/articles/Email-Communication-Readme.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+
+
+
+## Pack Workflow
+When an email is sent to the email address configured in your email integration, your email listener fetches the incoming email in Cortex XSOAR. 
+
+If the email is in response to an existing incident, an 8-digit number will appear next to the subject of the email, for example <93075875> Incident1.
+
+The pre-process script searches for the 8-digit number in the subject of the email to link to the incident.
```
Comment (omercnet) [inline content-repo/extra-docs/articles/Email-Communication-Readme.md:17]: The script does not search for 8-digit number, but with an `#` followed by an incident id, foe examples `#1234`

File: content-repo/extra-docs/articles/Email-Communication-Readme.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+### Layout
+There is 1 layout - **Email Communication** 
+
+There are 3 interactive sections in which you can specify 1 or more email addresses to add as a CC to the email, create the body of the email, and add attachments. The remaining 3 sections are for viewing the original email and all the email communications and attachments associated with the incident.
+
+
+
+![Layout](https://raw.githubusercontent.com/demisto/content/84e7bc89c8757544804540e6711d4b9aba210ec1/Packs/EmailCommunication/doc_files/Email_Communication_layout.png)
+
+You can use the layout as-is for email communication. It can also be used for new incident types by adding the **Email Communication** layout to an incident type. See [Add the Email Communication Layout to an Incident Type](#add-the-email-communication-layout-to-an-incident-type) for details.
```
Comment (omercnet) [inline content-repo/extra-docs/articles/Email-Communication-Readme.md:63]: This will not work easily - the Email Communication pack depends on many Email related fields, for it to work on other incident types those fields must be associated with the new Incident Type

File: content-repo/extra-docs/articles/Email-Communication-Readme.md [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+| Email thread | Displays the entire email thread including the original email and all email replies. |  
+| Original Email HTML | Displays in HTML format the original email that opened the incident. |  
+| Mail Attachments | Displays the metadata of the email attachments. Contains the Send Reply button.  |  
+ ---
+
+> Note: If an email cannot be sent to a specified address, no notification will appear in Cortex XSOAR.
+ 
+## Before You Start
+
+This pack requires that you must have active instances of both a mail listener and mail sender integration in order to send and receive emails. Configure either the [Gmail integration](https://xsoar.pan.dev/docs/reference/integrations/gmail)  or both the [EWS Mail Sender](https://xsoar.pan.dev/docs/reference/integrations/ews-mail-sender) and [EWS V2](https://xsoar.pan.dev/docs/reference/integrations/ews-v2) integrations. In addition, configure the Demisto REST API integration which requires a Demisto API key.
```
Comment (omercnet) [inline content-repo/extra-docs/articles/Email-Communication-Readme.md:84]:
```suggestion
This pack requires that you must have active instances of both a mail listener and mail sender integration in order to send and receive emails, and an active instance of Demisto REST API integration. Configure either the [Gmail integration](https://xsoar.pan.dev/docs/reference/integrations/gmail) or both the [EWS Mail Sender](https://xsoar.pan.dev/docs/reference/integrations/ews-mail-sender) and [EWS V2](https://xsoar.pan.dev/docs/reference/integrations/ews-v2) integrations. In addition, configure the Demisto REST API integration which requires a Demisto API key.
```

need xref for Demisto REST API docs

File: content-repo/extra-docs/articles/Email-Communication-Readme.md [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+To edit a layout, you must duplicate the layout and then edit the copy.
+  1. Navigate to **Settings -> Advanced -> Layouts**.
+  2. Select the layout you want to edit and click **Duplicate**.
+  3. Click the duplicate layout.
+  4. From the Library, select **Tabs** and search for **Email Communication**.
+  5. Drag and drop the Email Communication tab on the layout's tabs.
+  5. Click **Save**.
+
+
+**Customize the Send Reply Button**
```
Comment (omercnet) [inline content-repo/extra-docs/articles/Email-Communication-Readme.md:161]: it's not clear that this is a REQUIRED step in order to get the pack to work

**Review by omercnet — CHANGES_REQUESTED** [review]
few small changes

the `service_mail` parameter is a real pain here - suggest we find a way to not require duplicating the layout for this pack to work

File: content-repo/extra-docs/articles/Email-Communication-Readme.md (hunk 7 above)
Comment (altmannyarden) [reply content-repo/extra-docs/articles/Email-Communication-Readme.md:63]: Maybe we can add that It can also be used for new **email** related incident types

File: content-repo/extra-docs/articles/Email-Communication-Readme.md (hunk 6 above)
Comment (altmannyarden) [reply content-repo/extra-docs/articles/Email-Communication-Readme.md:17]: @omercnet We have an enhancement for that, will be release soon. https://github.com/demisto/content/pull/10222

**Comment by ShirleyDenkberg** [comment]
@omercnet @glicht @michalgold 
Omer's comments were implemented with two exceptions:
- Omer made the following comment: "the service_mail parameter is a real pain here - suggest we find a way to not require duplicating the layout for this pack to work".  Was that comment directed to the documentation? 
- "need xref for Demisto REST API docs"  Unfortunately, we do not have a ref for the Demisto REST API docs.

**Comment by michalgold** [comment]
@ShirleyDenkberg the API documentation itself is within the product, but I definitely agree that we need an article to better explain how to set it up.  We can also add this reference (https://docs.paloaltonetworks.com/cortex/cortex-xsoar/5-5/cortex-xsoar-admin/incidents/incidents-management/post-processing-for-incidents/create-a-post-processing-script.html) to how to set up pre-processing rule but even there we assume the customer knows how to configure Rest API. 
@kirbles19 maybe there should be a dedicated article for the section you wrote in health check pack?

**Comment by kirbles19** [comment]
> @kirbles19 maybe there should be a dedicated article for the section you wrote in health check pack?

@michalgold Which section specifically?

**Comment by altmannyarden** [comment]
> Omer made the following comment: "the service_mail parameter is a real pain here - suggest we find a way to not require duplicating the layout for this pack to work". Was that comment directed to the documentation? 

@ShirleyDenkberg  you are not require to duplicating the layout, you can deattch the sript, add a value for the service mail and attach it again

**Comment by ShirleyDenkberg** [comment]
@glicht @michalgold @omercnet @altmannyarden  
I added the post-processing link as Michal requested and I updated the instructions for configuring the service_email parameter.

**Comment by ShirleyDenkberg** [comment]
@glicht @michalgold @omercnet @altmannyarden Changed link to pre-processing rules.

File: content-repo/extra-docs/articles/Email-Communication-Readme.md (hunk 8 above)
Comment (altmannyarden) [reply content-repo/extra-docs/articles/Email-Communication-Readme.md]: There no Rest API docs yet, but we added an explanation of how to configure the demisto rest api
